### ISSUE TITLE
Add --parents option for COPY in Dockerfiles

### ISSUE THREAD

**Issue by uckelman** [containers/buildah#5557]
**Description**

Docker added a `--parents` option to the Dockerfile `COPY` command last year:

https://github.com/moby/moby/issues/35639
https://github.com/moby/buildkit/pull/3001
https://docs.docker.com/reference/dockerfile/#copy---parents

buildah doesn't know how to handle `COPY --parents` so errors out on Dockerfiles which use it.

I'm seeing increasing use of `--parents` in Dockerfiles I have to deal with, which is making it hard to use podman/buildah instead of Docker. `--parents` is also a genuinely useful option on its own, so would be nice to have not just for compatibility.

It's quite useful to be able to do things like
```
COPY --parents src/**/pyproject.toml src/**/poetry.lock /app/
```
and maintain the directory structure in the destination without having to list out every single file.

**Comment by TomSweeneyRedHat** [comment]
I believe @mheon is either working on this or looking into this.  Matt?

**Comment by mheon** [comment]
Yep, hoping to get to this over the next few weeks

**Comment by cyqsimon** [comment]
This is so badly needed with Java projects that involve multiple `pom.xml` files. Any progress on this?

**Comment by krmao** [comment]
any more update?

**Comment by jorlando-elucid** [comment]
bump this is a good idea

**Comment by Peneheals** [comment]
any news? :)

**Comment by rhatdan** [comment]
Could you give me a sample Containerfile that would use this?

**Comment by cyqsimon** [comment]
@rhatdan Of course. This sort of thing is useful in a workspace of any language where nested manifest files are involved.

[Here](https://github.com/rust-lang/cargo/issues/2644#issuecomment-2335736020) is an example in Rust (they are working on toolchain support for building only dependencies without project source, but that's not our concern).

Alternatively, [here](https://github.com/jabbrwcky/sample-maven-structure) is an example with Maven with existing toolchain support. To optimise layer caching, you would want something like this:

```dockerfile
COPY --parents **/pom.xml ./
RUN mvn dependency:go-offline
```

Without `--parents`, you'd have to manually path to every sub-manifest.

**Comment by oliversalzburg** [comment]
https://github.com/larixer/docker-js-monorepo has some pretty good examples as well.

**Comment by kikorb** [comment]
Any news on this? it would be really helpful to have `--parents` supported

**Comment by mheon** [comment]
@Honny1 might have time to work on this?

**Comment by Honny1** [comment]
I will take a look.
